### ISSUE TITLE
html_head signal not fired for non-event pages in backend

### ISSUE THREAD

**Issue by henryk** [pretix/pretix#2988]
### Problem and impact

I'm developing a plugin and want to include additional scripts and styles in the backend, on *all* pages. The documentation lists two signal with virtually identical description:

 * https://docs.pretix.eu/en/latest/development/api/general.html#pretix.control.signals.html_head for the `<head>`, and
 * https://docs.pretix.eu/en/latest/development/api/general.html#pretix.control.signals.html_page_start for the main contents

The issue is here https://github.com/pretix/pretix/blob/eeb6e11934db088a6016697bbfe9c985381d6d75/src/pretix/control/context.py#L83 and has been present ever since the signal was introduced in 2015.  Compared to that, the `html_page_start` signal is implemented in https://github.com/pretix/pretix/blob/eeb6e11934db088a6016697bbfe9c985381d6d75/src/pretix/control/templates/pretixcontrol/base.html#L379 and not subject to an `event` restriction.

### Expected behaviour

The wording "every page in the backend" in the documentation particularly suggests that these two signals would equally apply to, well, every page in the backend.
(Alternatively the documentation needs to highlight the difference. :)

### Steps to reproduce

1. Create a globally active plugin that implements `pretix.control.signals.html_head` and `pretix.control.signals.html_page_start`. Let the plugin return some dummy content.
2. Visit `/control/`: The `html_page_start` signal is fired and the content visible at the start of the main content.
3. Create an event and visit `/control/event/...`: The `html_page_start` *and* `html_ head` signals are fired. Content is visible both in page main and in `<head>`.

**Comment by raphaelm** [comment]
Indeed one of them is declared to be a plain Signal and one to be an EventPluginSignal. I can't recall why that happened but I'm also not sure how careful we need to be with changing this for compatibility

**Comment by henryk** [comment]
Oh, you're right. I hadn't even noticed that since (at this point) I don't care about the event. I agree that sending `None` for a parameter that was previously documented as "will contain the event" is a no-go. And inventing a DummyEvent (like Django's `AnonymousUser`) is probably overkill. A compromise would be a new signal only for non-event pages?

In the mean time I *do* have a workaround, but it's not fit for general consumption. I'm basing this on the fact that `html_head` is `""` in the event case.

<details>
<summary>Danger: Don't do this unless you're absolutely sure that you're the only plugin using this trick</summary>

In signals:

````python
@receiver(html_head)
def my_html_head(sender, request=None, **kwargs):
    return "bla"

def inject_head(request):
    # This hack overrides `html_head` in backend *only* if not in event context
    from django.urls import get_script_prefix
    if not request.path.startswith(get_script_prefix() + 'control') or not hasattr(request, 'user'):
        return {}
    if request.user.is_authenticated and not hasattr(request, 'event'):
        return {"html_head": my_html_head(None, request)}
    return {}
````

and in `PluginApp.ready()`:

````python
from django.conf import settings

settings.TEMPLATES[0]['OPTIONS']['context_processors'].append('...signals.inject_head')
````
</details>

**Comment by raphaelm** [comment]
Yeah, the underlying conceptional issue is that pretix historically only has a concept of plugins on event level and every plugin that operates on organizer and global level is basically *some* level of hack or workaround. We need to fix that at some point, but it's unclear how to do it without creating a hard-to-understand user experience.
